### ISSUE TITLE
Question on Policy Gradient

### ISSUE THREAD

**Issue by lyokka** [ageron/handson-ml#114]
Hello, I think policy gradient can also be used to solve continuous control problem, the main problem is how to get actions in continuous case, from other material, I learned it could be changed by a gaussian distribution, but I am not sure about the loss in this case. but after I simply change the code like this

original:
```python
outputs = tf.nn.sigmoid(logits)  # probability of action 0 (left)
p_left_and_right = tf.concat(axis=1, values=[outputs, 1 - outputs])
action = tf.multinomial(tf.log(p_left_and_right), num_samples=1)

y = 1. - tf.to_float(action)
```

changed:
```python
mu = tf.layers.dense(hidden, n_outputs)
action = mu + 0.001 * tf.random_normal([1, n_outputs])
loss = -tf.nn.l2_loss(action - mu)
```

It doesn't work well on "InvertedPendulum-v1", can anyone give me some suggestion on how to convert this code to continuous task, I really appreciate about that.

**Comment by ageron** [comment]
Hi @lyokka ,

Sorry for the long delay, I somehow missed your question.  Indeed, it is possible to use policy gradients to solve a continuous control problem.  The basic solution is to use a Gaussian Policy: the policy network predicts a point in the continuous action space (it predicts the mean action), then to select an actual action we just add some Gaussian noise to this mean, with a given variance (the variance would typically be a hyperparameter, but it is also possible to make the network learn to predict the right variance as well). Then you need to change the loss function, as explained in this great [lecture on Policy Gradients by David Silver](https://www.youtube.com/watch?v=KHZVXao4qXs&feature=youtu.be&t=2365).
I'll add a code example when I have more time (in the mean time, you can check out [this code](https://github.com/dennybritz/reinforcement-learning/blob/master/PolicyGradient/Continuous%20MountainCar%20Actor%20Critic%20Solution.ipynb)).
